Neos.ContentRepository.Core/Classes/Feature/NodeModification/NodeModification.php

```diff
@@ -112,7 +113,7 @@ private function handleSetSerializedNodeProperties(
                 $command,
                 Events::fromArray($events)
             ),
-            ExpectedVersion::ANY()
+            ExpectedVersion::fromVersion($contentStreamVersion->unwrap())
```

Comment:
This example shows how we can turn the constraint checks into actual hard constraints..
IMO we have to do this for all (content stream) events